apps/api/test/04-report-4.e2e-spec.ts

```diff
@@ -0,0 +1,265 @@
+      await getStatus(STATUS.JOB_OFFER_ACCEPTED),
+    );
+
+    statusTracker.push(acceptedMilestone.id);
+    const { body: afterJobAccept } = await request(app.getHttpServer()).get(URLS.REPORT4);
+
+    // With job offer accepted, BCCNM status should not be counted
+    expect(afterJobAccept).toStrictEqual(before);
+
+    for (let i = 0; i <= statusTracker.length - 1; i++) {
+      await deleteApplicantStatus(applicantIds[1], statusTracker[i]);
+    }
+  });
+
+  it('Applicant with all 4 licenses and check granted licensure', async () => {
+    for (const stat of BCCNM_LICENSE) {
+      const milestone = await addMilestone(applicantIds[2], '', await getStatus(stat));
+      const { body: afterLicense } = await request(app.getHttpServer()).get(URLS.REPORT4);
+      expect(reportFourNumberOfApplicants(afterLicense, stat)).toBe('1');
+      await deleteApplicantStatus(applicantIds[2], milestone.id);
+    }
+  });
+
+  it('NCAS value increases on report for each status Completed CBA or Completed SLA', async () => {
+    const { body: before } = await request(app.getHttpServer()).get(URLS.REPORT4);
+
+    //Expect the NCAS value to increase by 2
+    const expectedResult = before.map((item: { applicants: string; status: string }) => {
+      const stat = item.status;
+      let result = parseInt(item.applicants);
+      if (stat === STATUS.REFERRED_TO_NCAS) {
+        result += 2;
+      }
+      return {
+        ...item,
+        applicants: result.toString(),
+      };
+    });
+
+    const job1 = await addJob(applicantIds[3], {
+      ha_pcn: haPcnArray[2],
+      job_id: '344356',
+      recruiter_name: 'Test1',
+    });
+    const job2 = await addJob(applicantIds[4], {
+      ha_pcn: haPcnArray[3],
+      job_id: '45678',
+      recruiter_name: 'Test2',
+    });
+
+    const milestone1 = await addMilestone(
+      applicantIds[3],
+      job1.id,
+      await getStatus(STATUS.COMPLETED_CBA),
+    );
+    const milestone2 = await addMilestone(
+      applicantIds[4],
+      job2.id,
+      await getStatus(STATUS.COMPLETED_SLA),
+    );
+
+    const { body: after } = await request(app.getHttpServer()).get(URLS.REPORT4);
+
+    await deleteApplicantStatus(applicantIds[3], milestone1.id);
+    await deleteApplicantStatus(applicantIds[4], milestone2.id);
+
+    expect(expectedResult).toStrictEqual(after);
+  });
+
+  it('NNAS value increases on report for each status Received NNAS Report or Submitted documents', async () => {
+    const { body: before } = await request(app.getHttpServer()).get(URLS.REPORT4);
+
+    // Expect the NNAS value to increase by 2
+    const expectedResult = before.map((item: { applicants: string; status: string }) => {
+      const stat = item.status;
+      let result = parseInt(item.applicants);
+      if (stat === STATUS.APPLIED_TO_NNAS) {
+        result += 2;
+      }
+      return {
+        ...item,
+        applicants: result.toString(),
+      };
+    });
+
+    const job1 = await addJob(applicantIds[5], {
```

Comment:
A job is relevant to the recruitment milestones.

Reply:
> A job is relevant to the recruitment milestones.

Thanks for the clarification. I misunderstood this.